src/components/_templates/Graph/Graph.tsx

```diff
@@ -201,36 +202,35 @@ class Graph extends React.Component<Record<string, never>, Readonly<IGraphState>
   render() {
     return (
       <div className={GRAPH.GRAPH} onMouseDown={this.onMouseDown} onMouseUp={this.onMouseUp}>
-        <div className={GRAPH.BUTTONS}>
+        <Menu>
+          <Button href="/" asHref="/" type={BUTTON_TYPE.GREEN}>
+            {"< Back"}
+          </Button>
           <p className={GRAPH.TITLE}>{this.state.searching ? STATUS.SEARCHING : STATUS.CHOSE_ALGORITHM}</p>
-          <Button classNames={GRAPH.BUTTON} onClick={this.state.changed ? this.clear : this.drawMase}>
+          <Button onClick={this.state.changed ? this.clear : this.drawMase}>
             {this.state.changed ? "Clear Board" : "Draw Mase"}
           </Button>
           <Dropdown
             defaultValue={this.state.currentAlgorithm}
-            classNames={GRAPH.DROPDOWN}
             onChange={this.changeAlgorithm}
             list={Object.values(SUPPORTED_GRAPH_ALGORITMS)}
           ></Dropdown>
-          <Button classNames={GRAPH.BUTTON} onClick={this.search}>
-            Search
-          </Button>
-        </div>
-        <div className={GRAPH.BOX}>
-          {Object.values(this.state.matrix).map((row, rowIndex) =>
-            Object.values(row).map((vertex: IVertex, vertexIndex) => (
+          <Button onClick={this.search}>Search</Button>
+        </Menu>
+        <VisualBox type={VISUAL_BOX_TYPES.GRAPH}>
+          {Object.values(this.state.matrix).map((row, r) =>
```

Comment:
Try to avoid such one-letter variable names as `r` it is harder to understand and might be confusing

Reply:
You can also consider adding eslint rule [id-length](https://eslint.org/docs/rules/id-length) for this

Reply:
Add id-length rule with some exceptions like "j", "i" (for loops), "x" and "y" (name speaks for itself)